### ISSUE TITLE
Need clarification regarding SGD Optimizer

### ISSUE THREAD

**Issue by worldpeaceaspirer** [tensorflow/tensorflow#49461]
I have a question regarding `SGD Optimizer`. Same issue has been raised in [Stack Overflow](https://stackoverflow.com/questions/67636925/need-clarification-regarding-sgd-optimizer) as well.

There are 3 types of `Gradient Descent Algorithm`:

1. Batch Gradient Descent
2. Mini-Batch Gradient Descent and
3. Stochastic Gradient Descent

**`Stochastic Gradient Descent`** is an `Algorithm` in which one `Instance` from `Training Set` is taken at `Random` and the **`Weights`** are updated with respect to that `Instance`.

**`SGD Optimizer`** is slightly deviating from the above definition where it can accept the **`batch_size`** of more than 1. Can someone clarify this deviation?

Below code seems to be in line with the definition of **`Stochastic Gradient Descent`**:

    model.compile(optimizer = 'sgd', loss = 'mse')
    model.fit(x, y,epochs = 500, batch_size = 1,verbose=1)

However, below code seems to be confusing/deviating (since `batch_size` > 1):

    model.compile(optimizer = 'sgd', loss = 'mse')
    model.fit(x, y,epochs = 500, batch_size = 32, verbose=1)

Thank you in advance for the clarification.

**Comment by UsharaniPagadala** [comment]
@worldpeaceaspirer 
Could you please go through this [article](https://machinelearningmastery.com/how-to-control-the-speed-and-stability-of-training-neural-networks-with-gradient-descent-batch-size/), it helps

GitHub is mainly for addressing bugs in build/installation and performance.Keep track the issue posted in Stack overflow as you mentioned.There is a big community to support and learn from your questions. Thanks

**Comment by worldpeaceaspirer** [comment]
@UsharaniPagadala,
It's been answered in Stack Overflow. Thanks!
